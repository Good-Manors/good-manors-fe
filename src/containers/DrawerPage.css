.DrawerPageWrapper {
  width: 100vw;
  height: 100vh;
  display: grid;
  grid-template-columns: 25% 75%;
  grid-template-rows: 15% 85%;
  grid-template-areas: "header menu"
                       "header cardList";
                       
}

.CardListWrapper {
  display: flex;
  flex-direction: column;
  height: 100vh;
  align-content: center;
  overflow: auto;
  margin: 2em 1em;

  h2 {
    color: black;
  }
}
