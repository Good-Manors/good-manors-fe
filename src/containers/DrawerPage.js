import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import Header from '../components/Header/Header';
import Menu from '../components/Menu';
import styles from './DrawerPage.css';
import CardList from '../components/Card/CardList';
import { getFirstHome } from '../services/homes';
import { setHome } from '../actions/homeActions';

const DrawerPage = () => {

  const dispatch = useDispatch();

  const { drawerId } = useParams();
  console.log('params from drawerpage', drawerId);

  useEffect(() => {
    getFirstHome()
      .then(home => {
        dispatch(setHome(home));
      });
  }, []);

  return (
    <section id='drawer-page-wrapper' className={styles.DrawerPageWrapper}>
      <Header />
      <Menu />
      <section className={styles.CardListWrapper}>
        <CardList drawer={drawerId || ''} />
      </section>
    </section>
  );
};

export default DrawerPage;
