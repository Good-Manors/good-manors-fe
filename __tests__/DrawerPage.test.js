import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import DrawerPage from '../src/containers/DrawerPage';

describe('Renders a DrawerPage', () => {
  it('Should Render DrawerPage', () => {
    const getWrapper = (mockStore = createStore()) => mount(
      <Provider store={mockStore}>
        <DrawerPage/>
      </Provider>
    );
    expect(getWrapper).toMatchSnapshot();
  }); 
});
