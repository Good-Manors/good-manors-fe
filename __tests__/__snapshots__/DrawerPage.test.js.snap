// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a DrawerPage Should Render DrawerPage 1`] = `[Function]`;
